### ISSUE TITLE
Handling Non Existent LDAP User

### ISSUE THREAD

**Issue by FroeMic** [Adldap2/Adldap2-Laravel#329]
This is rather a **question** than a proper issue. I still think, a possible answer to this might be helpful for others as well.

I am creating sort of a CMS for teachers and want to use an existing LDAP System for Authentication. The coupling with the LDAP System should be as loose as possible. It could happen that a teacher wants to add information about their students, before these are actually created in the LDAP system. As a result I need to create a temporary User in my Users Table, to have a definite user id, with which this data is associated.
Later, when the students first login, certain LDAP properties (userprincipalname, email, ...) should be synced to my database. How can I make sure that the right row in the database is updated?

**Example**
```
        Schema::create('users', function (Blueprint $table) {
            $table->increments('id'); // should stay the same once created
            $table->string('email')->unique(); // can be added by the teacher before actually synced
            $table->string('userprincipalname')->unique(); // only synced with LDAP later
            // and many more ... 
        });
    }
```

1. DB State before Login: Table Users
```
ID        email                         userprincipalname
1          m.f@school.edu       null
```

2. Desired DB State after sync
```
ID        email                         userprincipalname
1          m.f@school.edu       m.f1234
```

**tl;dr:** How can I create users in my Users table manually **before the users are created in the LDAP**, so they have an associated id in my database. And how can I make sure, that when later syncing the user, the right entry is updated?

**Comment by stevebauman** [comment]
Hi @FroeMic,

Users are created and discovered by your `usernames` array in your `config/adldap_auth.php` file.

You just need to make sure that the user you create in the local database has the same email address as the LDAP account. Attributes will be synced upon the users first login when their LDAP account is found.

You can see how users are discovered and created in your local database but viewing the `Importer` class:

https://github.com/Adldap2/Adldap2-Laravel/blob/master/src/Auth/Importer.php#L12-L58

**Comment by FroeMic** [comment]
Thanks for pointing me into the direction. I will follow up once I have come up with a solution.
